src/AutoRest.CSharp/Mgmt/Decorator/PropertyMatchDetection.cs

```diff
@@ -32,11 +37,43 @@ internal static bool IsEqual(List<PropertyInfo> parentProperties, List<ObjectTyp
 
             foreach (var childProperty in childProperties)
             {
-                if (!DoesPropertyExistInParent(childProperty, parentDict, propertiesInComparison))
+                if (!DoesPropertyExistInParent(sourceType, targetType, childProperty, parentDict, propertiesInComparison))
+                {
+                    if (allowExtra)
+                        continue;
+
                     return false;
+                }
+
+                matchCount++;
             }
 
-            return true;
+            return matchCount == parentProperties.Count;
+        }
+
+        private static bool GetAllowSetting(Type sourceType)
+        {
+            var attribute = sourceType.GetCustomAttributes(false).FirstOrDefault(a => a.GetType().Name.Equals(TypeReferenceTypeChooser.TypeReferenceTypeAttributeType));
+            var allowExtraValue = attribute?.GetType().GetProperty("IgnoreExtraProperties", BindingFlags.Instance | BindingFlags.Public)?.GetValue(attribute);
+            return allowExtraValue is null ? false : (bool)allowExtraValue;
+        }
+
+        private static void AddInternalIncludes(Type sourceType, List<PropertyInfo> parentProperties)
+        {
+            var typeReferenceAttribute = sourceType.GetCustomAttributes(false).FirstOrDefault(a => a.GetType().Name.Equals(TypeReferenceTypeChooser.TypeReferenceTypeAttributeType));
+            if (typeReferenceAttribute is not null)
+            {
+                var internalToInclude = typeReferenceAttribute.GetType().GetProperty("InternalPropertiesToInclude", BindingFlags.Instance | BindingFlags.Public)?.GetValue(typeReferenceAttribute);
+                if (internalToInclude is not null)
+                {
+                    foreach (var propertyName in (string[])internalToInclude)
+                    {
+                        var property = sourceType.GetProperty(propertyName, BindingFlags.Instance | BindingFlags.NonPublic);
+                        if (property is not null)
```

Comment:
Should the property not existing trigger an error?

Reply:
Thought about that although the error will still happen from the lack of replacement.  I can throw a precise error here instead.